Firestore/Source/Local/FSTRemoteDocumentChangeBuffer.mm

```diff
@@ -47,42 +51,41 @@ + (instancetype)changeBufferWithCache:(id<FSTRemoteDocumentCache>)cache {
 - (instancetype)initWithCache:(id<FSTRemoteDocumentCache>)cache {
   if (self = [super init]) {
     _remoteDocumentCache = cache;
-    _changes = [NSMutableDictionary dictionary];
+    _changes = absl::make_unique<std::map<DocumentKey, FSTMaybeDocument *>>();
   }
   return self;
 }
 
 - (void)addEntry:(FSTMaybeDocument *)maybeDocument {
   [self assertValid];
 
-  self.changes[(FSTDocumentKey *)maybeDocument.key] = maybeDocument;
+  _changes->emplace(maybeDocument.key, maybeDocument);
```

Comment:
This isn't quite the same. Consider:

```
addEntry(Document(key="abc", otherstuff="def");
addEntry(Document(key="abc", otherstuff="123");
```

In the old code, changes would contain the doc with otherstuff=123, but in the new code it'll contain the doc with otherstuff=def. (emplace/insert are effectively noops if the key already exists in the map.)

`changes[key] = maybeDoc;` should more closely follow the existing implementation.

(That said, I'm not sure if it's possible to get into this situation, so maybe this doesn't matter.)